src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,90 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    List<String> tests = new ArrayList<String>();
+    String testngXmlAsString;
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {
+        verifyTestngXmlExists();
+        readTestngAsString();
+        
+        getTestFiles();
+        excludeFilesNotInTestng();
+
+        verifyTestngContainsTests();
```

Comment:
This approach to coding has low testability. For better testability, a method should get all it needs as parameters and return all it creates as a return value, as far as possible. For example, `readTestngAsString()` should return a string. SLAP is more than simply extracting chunks of code into methods.